craft_application/services/base.py

```diff
@@ -41,5 +46,32 @@ def __init__(self, app: AppMetadata, services: ServiceFactory) -> None:
         self._services = services
 
     def setup(self) -> None:
-        """Application-specific service setup."""
+        """Service-specific setup to perform.
+
+        Child classes should always call ``super().setup()``.
+        """
         emit.debug(f"Setting up {self.__class__.__name__}")
+
+    @property
+    def _project(self) -> Project:
```

Comment:
Maybe that was on purpose on you missed one call to `self._services.get("project").get()`  https://github.com/canonical/craft-application/blob/16a20e36ac67c17f78f90e17402a96610d1ab34f/craft_application/commands/remote.py#L142

Reply:
That's on a command (separate PR) :-)

Reply:
Specifically this one: https://github.com/canonical/craft-application/pull/912